src/trade-params/TradeParams.js

```diff
@@ -101,8 +115,8 @@ export default class TradeParams extends Component {
     }
 
     componentWillUnmount() {
-        const { proposal, proposalError } = this.props;
-        if (proposalError) {
+        const { proposal, errors } = this.props;
+        if (errors.proposalError) {
```

Comment:
You probably can abstract away these in a lib...
Like canUpdateProposal() or something.
Didn't think it through much, but basically the biz logic if possible to go outside of the component itself.